lib/game_runner.rb

```diff
@@ -0,0 +1,146 @@
+class GameRunner
+
+    def initialize
+        @player = nil
+        @game = nil
+    end
+
+    def welcome
+        puts `clear`
+        puts "Let's play Crazy Eights! Enter your username: "
+        name = STDIN.gets.strip.downcase.capitalize
+        player = Player.find_by username: name
+        if player.nil?
+            @player = Player.create(username: name)
+            puts "Welcome, new player #{name}!"
+        else
+            @player = player
+            puts "Welcome back #{name}!"
+        end
+    end
+
+    def game_options
+        user_is_active = true
+        while user_is_active
+            input = home_menu
+            puts `clear`
+            case input
+            when '1'
+                set_up
+                play_crazy_eights(false)
+            when '2'
+                set_up
+                @game.deal_start_hand('computer')
+                play_crazy_eights(true)
+            when '3' then rules
+            when '4'
+                puts 'Goodbye!'
+                user_is_active = false
+            else
+                puts "Mmm, I'm getting mixed signals. Can you try making a "
+                puts 'selection again? Just enter 1, 2, or 3 please.'
+            end
+        end
+    end
+
+    def home_menu
+        puts
+        puts "To start a single player game, [enter] '1'"
+        puts "To play against the computer, [enter] '2'"
+        puts "To view the rules, [enter] '3'"
+        puts "To exit, [enter] '4'"
+        puts
+        input = STDIN.gets.chomp.strip
+        input
+    end
+
+    def set_up
+        @game = CrazyEightGame.create(player_id: @player.id, turn_count: 0)
+        @game.new_deck
+        @game.deal_start_hand(@player.id)
+        @game.place_start_card
+    end
+
+    def game_over?(computer)
+        @game.nil? || @game.remaining.zero? || @game.player_hand(@player.id).length.zero? || computer && @game.player_hand('computer').length.zero? 
+    end
+
+    def play_crazy_eights(computer)
+        until game_over?(computer)
+            if computer && @game.turn_count % 2 == 1 
+                @game.computer_turn
+            else
+                @game.view_top_card
+                @game.view_hand(@player.id)
+                choice = turn_options.upcase
```

Comment:
```suggestion
                print_turn_options
                choice = STDIN.gets.strip.upcase
```
Try and handle the input seperate from the prints. Also do the upcas either when doing the check, or best yet when getting the input.